tests/pos/i21154/Z.tastycheck

```diff
@@ -0,0 +1,1340 @@
+  1045: 326 .. 326
+  1050: 326 .. 326
+  1055: 326 .. 326
+  1067: 326 .. 326
+  1086: 321 .. 321
+  1090: 321 .. 321
+  1098: 321 .. 321
+  1103: 336 .. 336
+  1106: 336 .. 336
+  1109: 336 .. 336
+  1115: 321 .. 336
+  1118: 321 .. 321
+  1139: 321 .. 336
+  1143: 321 .. 321
+  1152: 321 .. 321
+  1157: 321 .. 321
+  1163: 321 .. 321
+  1168: 321 .. 321
+  1172: 321 .. 321
+  1175: 326 .. 326
+  1179: 326 .. 326
+  1188: 326 .. 326
+  1194: 326 .. 326
+  1200: 321 .. 321
+  1204: 321 .. 321
+  1212: 321 .. 321
+  1217: 321 .. 321
+  1220: 321 .. 321
+  1223: 321 .. 321
+  1227: 321 .. 321
+  1228: 321 .. 321
+  1230: 321 .. 321
+  1233: 321 .. 321
+  1236: 321 .. 321
+  1240: 326 .. 326
+  1243: 326 .. 326
+  1251: 326 .. 326
+  1254: 326 .. 326
+  1257: 326 .. 326
+  1260: 326 .. 326
+  1273: 326 .. 326
+  1280: 339 .. 354
+  1284: 352 .. 354
+  1293: 354 .. 354
+  1299: 352 .. 354
+  1303: 352 .. 352
+  1306: 344 .. 344
+  1310: 344 .. 344
+  1315: 344 .. 344
+  1319: 344 .. 344
+  1325: 344 .. 344
+  1328: 344 .. 344
+  1331: 344 .. 344
+  1334: 344 .. 344
+  1348: 344 .. 344
+  1359: 344 .. 344
+  1365: 344 .. 344
+  1377: 344 .. 344
+  1384: 344 .. 344
+  1402: 344 .. 344
+  1406: 344 .. 344
+  1409: 344 .. 344
+  1414: 344 .. 344
+  1417: 344 .. 344
+  1421: 344 .. 344
+  1426: 344 .. 344
+  1430: 344 .. 344
+  1436: 344 .. 344
+  1439: 344 .. 344
+  1442: 344 .. 344
+  1445: 344 .. 344
+  1453: 344 .. 344
+  1459: 344 .. 344
+  1477: 344 .. 344
+  1480: 344 .. 344
+  1483: 344 .. 344
+  1487: 344 .. 344
+  1490: 344 .. 344
+  1493: 344 .. 344
+  1497: 344 .. 344
+  1500: 344 .. 344
+  1503: 344 .. 344
+  1506: 344 .. 344
+  1511: 344 .. 344
+  1516: 344 .. 344
+  1528: 344 .. 344
+  1547: 344 .. 344
+  1550: 344 .. 344
+  1553: 344 .. 344
+  1556: 344 .. 344
+  1561: 344 .. 344
+  1566: 344 .. 344
+  1578: 344 .. 344
+  1597: 339 .. 339
+  1601: 339 .. 339
+  1609: 339 .. 339
+  1614: 354 .. 354
+  1617: 354 .. 354
+  1620: 354 .. 354
+  1626: 339 .. 354
+  1629: 339 .. 339
+  1650: 339 .. 354
+  1654: 339 .. 339
+  1663: 339 .. 339
+  1668: 339 .. 339
+  1674: 339 .. 339
+  1679: 339 .. 339
+  1683: 339 .. 339
+  1686: 344 .. 344
+  1690: 344 .. 344
+  1699: 344 .. 344
+  1705: 344 .. 344
+  1711: 339 .. 339
+  1715: 339 .. 339
+  1723: 339 .. 339
+  1728: 339 .. 339
+  1731: 339 .. 339
+  1734: 339 .. 339
+  1738: 339 .. 339
+  1739: 339 .. 339
+  1741: 339 .. 339
+  1744: 339 .. 339
+  1747: 339 .. 339
+  1751: 344 .. 344
+  1754: 344 .. 344
+  1762: 344 .. 344
+  1765: 344 .. 344
+  1768: 344 .. 344
+  1771: 344 .. 344
+  1784: 344 .. 344
+  1791: 354 .. 354
+  1794: 354 .. 354
+  1797: 354 .. 354
+  1800: 354 .. 354
+  1810: 354 .. 354
+  1822: 354 .. 354
+  1832: 354 .. 354
+  1839: 298 .. 298
+  1842: 298 .. 298
+  1848: 298 .. 298
+  1851: 298 .. 298
+  1854: 298 .. 298
+  1859: 298 .. 298
+  1864: 298 .. 298
+  1874: 293 .. 354
+  1880: 293 .. 293
+  1884: 293 .. 293
+
+  source paths:
+     0: 27 [<elided source file name>]
```

Comment:
the magic is that if you revert the change to Namer, then this will have three source paths - because the positions of the child annotations will be set in the context of `A.scala` - i.e. the call site that forces the completion